charmcraft.yaml

```diff
@@ -1,14 +1,15 @@
-# Copyright 2021 Canonical Ltd.
+# Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-type: "charm"
+type: charm
 bases:
   - build-on:
-      - name: "ubuntu"
-        channel: "20.04"
+    - name: "ubuntu"
+      channel: "20.04"
     run-on:
-      - name: "ubuntu"
-        channel: "20.04"
+    - name: "ubuntu"
+      channel: "20.04"
 parts:
   charm:
-    charm-python-packages: [setuptools, pip]  # Fixes install of some packages
+    charm-python-packages: [setuptools, pip]
+    build-packages: [cargo, rustc, pkg-config, libffi-dev, libssl-dev]
```

Comment:
What are these needed for?  This looks familiar but I can't think of why it would be needed now in this PR

Reply:
Yes

Reply:
Should we add a comment saying that?  I'm a bit torn, but I think a comment would help.  I doubt anyone will ever realize they can remove these if we stopped using pydantic